shared/models/git_repo.rb

```diff
@@ -293,43 +309,45 @@ def git_action_with_queue(ensure_block: nil, &block)
     # Discard any changes
     def reset_hard!
       git_action_with_queue do
-        super_verbose("starting reset_hard! #{self.git_config.git_url}".freeze)
+        logger.debug("starting reset_hard! #{self.git_config.git_url}".freeze)
         self.git.reset_hard
         self.git.clean(force: true, d: true)
-        super_verbose("done reset_hard! #{self.git_config.git_url}".freeze)
+        logger.debug("done reset_hard! #{self.git_config.git_url}".freeze)
       end
     end
 
     def fetch
       git_action_with_queue(ensure_block: proc { unset_auth }) do
-        super_verbose("starting fetch #{self.git_config.git_url}".freeze)
+        logger.debug("starting fetch #{self.git_config.git_url}".freeze)
         self.temporary_storage_path = self.setup_auth(repo_auth: repo_auth)
         self.git.fetch
-        super_verbose("done fetching #{self.git_config.git_url}".freeze)
+        logger.debug("done fetching #{self.git_config.git_url}".freeze)
       end
     end
 
     def clone(repo_auth: self.repo_auth, async: false)
       if async
         # If we're async, just push it on the queue
         git_action_with_queue(ensure_block: proc { unset_auth }) do
-          super_verbose("starting clone_synchronously of #{self.git_config.git_url}".freeze)
+          logger.debug("starting clone_synchronously of #{self.git_config.git_url}".freeze)
           clone_synchronously(repo_auth: repo_auth)
-          super_verbose("done clone_synchronously of #{self.git_config.git_url}".freeze)
+          logger.debug("done clone_synchronously of #{self.git_config.git_url}".freeze)
         end
       else
         clone_synchronously(repo_auth: repo_auth)
       end
     end
 
-    private
+    def callback_block(async_start)
+      # How do we know that the task was successfully finished?
+      return if self.callback.nil?
```

Comment:
It looks like if `self.callback` exists and `async_start` if false, we return without calling anything. 
Maybe we should crash in development mode since this shouldn't happen
Also, maybe we should check at initialization time for this bad state and error out there too.
Maybe in `validate_initialization_params`

Reply:
With the same thought, if we don't provide any callback when `async_start` we should raise another error?

Reply:
I think it's ok to not have a callback if you want to fire and forget, but I don't have a strong opinion one way or the other.

Reply:
:+1: !